File: String/Count and Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        def rle(s: str) -> str:
            res = ""
            curr_ch = s[0]
            curr_cnt = 0
            for char in s:
                if char == curr_ch:
                    curr_cnt+=1
                else:
                    res += f"{curr_cnt}{curr_ch}"
                    curr_ch = char
                    curr_cnt = 1
            res += f"{curr_cnt}{curr_ch}"
            return res

        curr = ""

        for i in range(1, n+1):
            if i == 1:
                curr = "1"
            else:
                curr = rle(curr)
        return curr

File: String/Count and Say.rs
impl Solution {
    pub fn count_and_say(n: i32) -> String {
        let mut curr = String::from("1");
        for _ in 2..=n {
            curr = Self::rle(&curr);
        }
        curr
    }

    fn rle(s: &String) -> String {
        let mut res = String::new();

        let mut curr_char = s.chars().nth(0).unwrap();
        let mut curr_cnt = 0;

        for ch in s.chars() {
            if ch == curr_char {
                curr_cnt += 1;
            } else {
                res.push_str(&curr_cnt.to_string());
                res.push(curr_char);
                curr_char = ch;
                curr_cnt = 1;
            }
        }
        res.push_str(&curr_cnt.to_string());
        res.push(curr_char);
        res
    }
}